src/status_im2/common/scroll_page/view.cljs

```diff
@@ -0,0 +1,76 @@
+(ns status-im2.common.scroll-page.view
+  (:require
+   [react-native.core :as rn]
+   [quo2.core :as quo]
+   [utils.re-frame :as rf]
+   [quo2.foundations.colors :as colors]
+   [status-im2.common.scroll-page.style :as style]
+   [oops.core :as oops]
+   [reagent.core :as reagent]
+   [quo.platform :as platform]))
+
+(defn icon-color []
+  (colors/theme-colors
+   colors/white-opa-40
+   colors/neutral-80-opa-40))
+
+(defn get-platform-value [value] (if platform/ios? (+ value 44) value))
+(def scroll-0 (if platform/ios? -44 0))
+(def scroll0 (if platform/ios? 44 0))
+(def scroll1 (if platform/ios? 86 134))
+(def scroll2 (if platform/ios? -26 18))
+
+(defn get-header-size [scroll-height]
+  (if (<= scroll-height scroll2)
+    0
+    (->>
+     (+ (get-platform-value -17) scroll-height)
+     (* (if platform/ios? 3 1))
+     (max 0)
+     (min (if platform/ios? 100 124)))))
+
+(def max-image-size 80)
+(def min-image-size 32)
+
+(defn diff-with-max-min [value maximum minimum]
+  (->>
+   (+ value scroll0)
+   (- maximum)
+   (max minimum)
+   (min maximum)))
+
+(defn scroll-page [icon cover page-nav name]
+  (let [scroll-height (reagent/atom scroll-0)]
+    (fn [sticky-header children]
+      [:<>
+       [:<>
+        [rn/image
+         {:source   cover
+          :position :absolute
+          :style    (style/image-slider (get-header-size @scroll-height))}]
+        [rn/blur-view (style/blur-slider (get-header-size @scroll-height))]]
+       [rn/view {:style {:z-index 6 :margin-top (if platform/ios? 56 12)}}
+        [quo/page-nav
+         {:horizontal-description?            true
+          :one-icon-align-left?               true
+          :align-mid?                         false
+          :page-nav-color                     :transparent
+          :page-nav-background-uri            ""
+          :mid-section {:type  :text-with-description
+                        :main-text (when (>= @scroll-height scroll1) name)
+                        :description-img (when (>= @scroll-height scroll1) icon)}
+          :right-section-buttons (:right-section-buttons page-nav)
+          :left-section {:icon                  :i/close
+                         :icon-background-color (icon-color)
+                         :on-press #(rf/dispatch [:navigate-back])}}]
+        (when sticky-header [sticky-header @scroll-height])]
+       [rn/scroll-view {:style (style/scroll-view-container (diff-with-max-min @scroll-height 16 0))
+                        :shows-vertical-scroll-indicator false
+                        :scroll-event-throttle 1
```

Comment:
I will play around with this some more and see how it works. At the moment the scrolling looks pretty good but there might be some more optimisations needed especially as we add more network requests to that page.. e.g get channel notifications etc

Reply:
I believe it works fine with 1ms, but it might be putting unnecessary stress on the system. According to the docs, the throttle parameter is there to help us find the sweet spot where scrolling performs well, but it's not wasting cycles too.

Reply:
ok, I tried at 16 and it's fine. If this is too little then we can increase this rate

Reply:
Thanks for putting in the effort to test this @J-Son89